src/pyJiraCli/cmd_search.py

```diff
@@ -142,73 +164,94 @@ def _cmd_search(filter_str: str, results: int, save_file: str, server: Server) -
 
-                    # Get the worklogs for the issue
-                    # Itereate over all worklogs and store them in a list
-                    worklog_list = [
-                        log.raw for log in jira.worklogs(issue.key)]
+                # Get the worklogs for the issue
+                # Itereate over all worklogs and store them in a list
+                worklog_list = [
+                    log.raw for log in jira.worklogs(issue.key)]
 
-                    issue_dict["fields"]["worklog"] = {
-                        "total": len(worklog_list),
-                        "worklogs": worklog_list
-                    }
+                issue_dict["fields"]["worklog"] = {
+                    "total": len(worklog_list),
+                    "worklogs": worklog_list
+                }
 
-                issue_list.append(issue_dict)
+            issue_list.append(issue_dict)
 
-            search_dict['issues'] = issue_list
+        search_dict['issues'] = issue_list
 
+        if save_file is not None:
             ret_status = _save_search(save_file, search_dict)
-
         else:
-            _print_table(found_issues)
+            _print_table(search_dict, fields)
+            ret_status = Ret.CODE.RET_OK
 
     return ret_status
 
 
-def _print_table(issues: list) -> None:
-    """ Print a quick overview for all issues in the list.
+def _print_table(search_dict: dict, fields: list[str]) -> None:
+    """ Print a quick overview for all issues in the dict.
 
     Args:
-        issues (list): list with all found issues
+        search_dict (dict): dict with all found issues and search metadata.
     """
-    for header in HEADER:
-        print(f"{header:<{HEADER_COL_WIDTH[header]}}", end="")
+
+    if fields == []:
+        fields = DEFAULT_FIELDS
+
+    issues = search_dict['issues']
+    number_of_fields = min(len(fields), MAX_FIELDS_PRINTED)
+
+    print(f"{'Key':<{COLUMN_WIDTH}}", end="")
+
+    for field in range(number_of_fields):
+        print(f"{fields[field]:<{COLUMN_WIDTH}}", end="")
     print()
 
     for issue in issues:
-        print(f"{issue.key:<{HEADER_COL_WIDTH['Key']}}", end="")
-        print(
-            f"{issue.fields.project.key:<{HEADER_COL_WIDTH['Project']}}", end="")
-        # pylint: disable=line-too-long
-        print(f"{issue.fields.summary[:HEADER_COL_WIDTH['Summary'] - 2]:<{HEADER_COL_WIDTH['Summary']}}",
-              end="")
-        print(
-            f"{issue.fields.created[:10]:<{HEADER_COL_WIDTH['Created']}}", end="")
-        print(
-            f"{issue.fields.creator.name:<{HEADER_COL_WIDTH['Creator']}}", end="\n")
+        issue_key = issue['key']
+        print(f"{issue_key:<{COLUMN_WIDTH}}", end="")
+
+        for key in range(number_of_fields):
+            field = fields[key]
+            field_value = issue["fields"][field]
+
+            if not isinstance(field_value, str):
+                field_value = "Complex Type:" + type(field_value).__name__
```

Comment:
is it really desired, that the table now shows `Complex Type:dict` for the fields `creator` and `project` when I use the default fields?

Reply:
If we start handling every complex field, then the tool stops being generic. But I will try to do something for such fields